### PR TITLE
Add version catalog potential breaking changes and improve version ca…

### PR DESCRIPTION
**Pull request by asodja**
…talog docs

Fixes #?

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test QFL

**Comment by asodja** [comment]
`Version catalog also offers programmatic access through the unsafe API. Unsafe API is provided by the version catalog extension:` does this sound weird?

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc [hunk 1]
```diff
@@ -284,14 +284,14 @@ or
 See the https://toml.io[TOML specification] for details.
 ====
 
-[[sub:programmatic-access-to-catalog]]
-=== Programmatic access to catalog
+[[sub:type-unsafe-access-to-catalog]]
+=== Type unsafe API
 
-You can also access version catalog programmatically with the version catalog extension and API:
+Version catalog also offers programmatic access through the unsafe API. Unsafe API is provided by the version catalog extension:
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc:290]: What about:
> Version catalogs can be accessed through a type unsafe API. This API is available in situations where generated accessors are not. It is accessed through the version catalog extension:

**Review by ljacomet — COMMENTED** [review]
Suggested edit for the documentation section:

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc (hunk 1 above)
Comment (asodja) [reply subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc]: Sounds better yes.

Comment (asodja) [reply subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc]: Fixed

**Review by ljacomet — APPROVED** [review]

**Comment by asodja** [comment]
@bot-gradle test and merge

**Pull request merged**